theano/gof/op.py

```diff
@@ -966,7 +966,7 @@ def make_thunk(self, node, storage_map, compute_map, no_recycling):
         if new_node is not None:
             node = new_node
 
-        if self._op_use_c_code:
```

Comment:
I don't like that fix. It request all \__getstate__ \__setstate__ to know the internal representation of ops.

An alternative is to add a method to Op that would do that and call it from all Op.\__setstate__. As there isn't other attribute from Op directly, it would be less error prone to fix it there. With your suggestion or my alternative, we need to verify all \__getstate__ and \__setstate__ in Theano and outside Theano to make sure they work well. This isn't possible. Fixing it here is a general fix. I already checked the other place we use it and it is guarded by a getattr() too. So it shouldn't pop up elsewhere.